dio/lib/src/options.dart

```diff
@@ -82,9 +82,9 @@ typedef RequestEncoder = List<int> Function(
 class BaseOptions extends _RequestConfig with OptionsMixin {
   BaseOptions({
     String? method,
-    int? connectTimeout,
-    int? receiveTimeout,
-    int? sendTimeout,
+    Duration? connectTimeout,
+    Duration? receiveTimeout,
+    Duration? sendTimeout,
```

Comment:
Hm, I'm not sure about that because by default [requests in web have no timeout](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/timeout). The only way to represent this is by making it nullable.

The receive and send timeout are also a concept which doesn't exist on web at all. So it's just bolted on by dio itself.

However after looking at it, it seems like the lines
https://github.com/flutterchina/dio/blob/develop/dio/lib/src/adapters/browser_adapter.dart#L66-L82 
aren't needed at all, if one just set the xhr.timeout. I'll have to test that.

Darts [`HttpClient`](https://api.dart.dev/stable/2.15.1/dart-io/HttpClient/connectionTimeout.html) also is nullable by default.

Reply:
Yea you are right.

> However after looking at it, it seems like the lines
develop/dio/lib/src/adapters/browser_adapter.dart#L66-L82
aren't needed at all, if one just set the xhr.timeout. I'll have to test that.

That would be great

Reply:
Lets do this separately